app/src/main/java/com/example/capstoneproject/view/SignInFragment.kt

```diff
@@ -27,10 +32,58 @@ class SignInFragment : Fragment() {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        binding.createAccount.setOnClickListener {
-            val action = SignInFragmentDirections.actionSignInFragmentToSignUpFragment()
-            findNavController().navigate(action)
+        observeData()
+        setupViews()
+    }
+
+    fun observeData() {
+        authViewModel.authResult.observe(viewLifecycleOwner, Observer { result ->
+            result.let {
+                if (it != null) {
+                    if (it.isSuccess) {
```

Comment:
Ayrıca söyle bir durum var kullanıcı succes aldığında bir sonraki ekrana gidecek burda geri gelemediği bir ekran diye düşünüyorum ama geri gelebildiği senaryoda liveData hala success olduğu için yine navigasyon işlemi yapacaktır. Burası için LiveDatayı tek seferlik kullanım şeklinde bir araştırabilirsin ya da her iki durumda da navigate işleminden sonra livedatayı null a çekmek olabilir kısa yoldan. Çok iyi bir yöntem mi tartışılır tabii.